.CMake/compiler_opts.cmake

```diff
@@ -170,6 +171,7 @@ elseif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
         set(THREADS_PREFER_PTHREAD_FLAG ON)
         find_package(Threads REQUIRED)
         set(OQS_USE_PTHREADS_IN_TESTS 1)
+        set(OQS_USE_PTHREADS_IN_LIBOQS 1)
```

Comment:
Same question wrt `gcc`. What about rather checking for presence of `pthreads` (e.g., as per https://stackoverflow.com/questions/1620918/cmake-and-libpthread)?